### PR TITLE
record metrics for deactivations

### PR DESCRIPTION
**Pull request by xcoulon**
fixes #ODC622

Signed-off-by: Xavier Coulon <xcoulon@redhat.com>

* Please describe what your change is about.
* What issues does it solve? (Use [autoreferencing for this](https://help.github.com/articles/basic-writing-and-formatting-syntax/#referencing-issues-and-pull-requests))
* If you are still working on the change please prefix the pull request title with "WIP"

### CODE REVIEW

**Comment by alien-ike** [comment]
### Ike Plugins (test-keeper)

Thank you @xcoulon for this contribution!



It appears that no tests have been added or updated in this PR.

Automated tests give us confidence in shipping reliable software. Please add some as part of this change.

If you are an admin or the reviewer of this PR and you are sure that no test is needed then you can use the command `/ok-without-tests` as a comment to make the status green.


Your plugin configuration is stored in the [file](https://github.com/fabric8-services/fabric8-auth/blob/bd6aadb053c461aabe68d7ed10b389833570bd5a/.ike-prow/test-keeper.yaml).

File: metric/metric.go [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+var (
+	userDeactivationNotificationCounter = prometheus.NewCounterVec(prometheus.CounterOpts{
+		Name: userDeactivationNotificationCounterName,
+		Help: "Total number of the users notified for upcoming deactivation",
+	}, []string{"successful", "failure"})
+	userDeactivationCounter = prometheus.NewCounterVec(prometheus.CounterOpts{
+		Name: userDeactivationCounterName,
+		Help: "Total number of the deactivated users",
+	}, []string{"successful", "failure"})
```
Comment (MatousJobanek) [inline metric/metric.go]: Is the `failure` label used anywhere?

File: metric/metric.go [hunk 2]
```diff
@@ -0,0 +1,59 @@
+package metric
+
+import (
+	"github.com/fabric8-services/fabric8-common/log"
```
Comment (sbryzak) [inline metric/metric.go]: Shouldn't we use the fabric-auth log package here?

File: metric/metric.go (hunk 1 above)
Comment (xcoulon) [reply metric/metric.go]: good point, I can certainly remove it.

File: metric/metric.go (hunk 2 above)
Comment (xcoulon) [reply metric/metric.go]: good point, @sbryzak

Comment (xcoulon) [reply metric/metric.go]: fixed in 5d6a0b4

**Comment by xcoulon** [comment]
thanks for the review @MatousJobanek !
@alexeykazakov I've added a test for the `/metrics` endpoint, similar to what @MatousJobanek did in tenant service: https://github.com/fabric8-services/fabric8-auth/pull/835/commits/510dd494cf800854266fb6334cce4d1d70fbb1e4

**Pull request merged**